addons/website_slides/static/src/js/slides_course_quiz.js

```diff
@@ -740,6 +739,49 @@
         // Private
         //---------------------------------------------------------------------
 
+        _getSlide: function (slideId) {
+            const slide = this._super(...arguments);
+            if (slide) {
+                return slide;
+            }
+            // A quiz in a documentation on non fullscreen view
+            return $(`.o_wslides_js_lesson_quiz[data-id="${slideId}"`).data();
+        },
+
+        /**
+         * After a slide has been marked as completed / uncompleted, update the state
+         * of this widget and reload the slide if needed (e.g. to re-show the questions
+         * of a quiz).
+         */
+        setSlideCompleted: function (slide, completed = true) {
+            this._super(...arguments);
+
+            if (this.quiz && this.quiz.slide.id === slide.id && !completed && this.quiz.quiz.questionsCount) {
+                // The quiz has been marked as "Not Done", re-load the questions
+                this.quiz.quiz.answers = null;
+                this.quiz.quiz.sessionAnswers = null;
+                this.quiz.slide.completed = false;
+                this.quiz._fetchQuiz().then(() => {
+                    this.quiz.renderElement();
+                    this.quiz._renderValidationInfo();
+                });
+
+            }
+
+            // The quiz has been submitted in a documentation and in non fullscreen view,
+            // should update the button "Mark Done" to "Mark To Do"
+            const $doneButton = $('.o_wslides_done_button');
+            if ($doneButton.length && completed) {
```

Comment:
Looks like this button shares a lot with the other button.
To see if same logic with a small external widget can be applied.
(Or maybe the same widget if there is a lot in common?).